### PR TITLE
custom style implement

### PR DESCRIPTION
**Pull request by ahmetsaridogan**
default styles may change privately. for example border width, background-color, etc.

Screen Shot 2022-01-19 at 16 34 53

### CODE REVIEW

**Review by ermeydan — APPROVED** [review]

File: src/utils.ts [hunk 1]
```diff
@@ -4,7 +4,18 @@ export interface BoundingBox {
   width: number;
   height: number;
 }
-
+export type ElementOverlayOptions = {
+  className?: string;
+  background?: string;
```
Comment (hmarr) [inline src/utils.ts:9]: I could see the style attributes expanding over time, so I think it'd be sensible to nest them in a `style` field, e.g.:

```typescript
export interface ElementOverlayStyleOptions {
  background?: string;
  ...
}

export interface ElementOverlayOptions {
  className?: string;
  style?: ElementOverlayStyleOptions;
}
```

**Review by hmarr — COMMENTED** [review]
Thanks for this PR and sorry for neglecting it!

This looks great — I just had one small comment about the structure of the options object, then I'll get it merged and released.

File: src/utils.ts (hunk 1 above)
Comment (ahmetsaridogan) [reply src/utils.ts:9]: hello again sorry it's been a long time.
thanks, you can review

File: example/example.js [hunk 2]
```diff
@@ -17,7 +17,10 @@ function main() {
       .join(" > ");
   };
 
-  const picker = new ElementPicker();
+  const picker = new ElementPicker({
+    background: "rgba(153, 235, 255, 0.5)",
+    borderColor: "yellow"
+  });
```
Comment (hmarr) [inline example/example.js]: I think this needs to be nested under a `style` attribute:

```suggestion
  const picker = new ElementPicker({
    style: {
      background: "rgba(153, 235, 255, 0.5)",
      borderColor: "yellow"
    }
  });
```

File: src/element-overlay.ts [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
-    this.overlay.style.borderColor = "#F95738";
-    this.overlay.style.borderStyle = "solid";
-    this.overlay.style.borderRadius = "1px";
-    this.overlay.style.borderWidth = "1px";
-    this.overlay.style.boxSizing = "border-box";
-    this.overlay.style.cursor = "crosshair";
-    this.overlay.style.position = "absolute";
-    this.overlay.style.zIndex = "2147483647";
+    this.overlay.className = options.className || "_ext-element-overlay";
+    this.overlay.style.background = options.background || "rgba(250, 240, 202, 0.2)";
```
Comment (hmarr) [inline src/element-overlay.ts]: I think all these lines need to use `options.style.{property}` (although you might want to extract `style` to its own variable to above). It looks like is currently causing the [build to fail](https://github.com/hmarr/pick-dom-element/runs/7119295345?check_suite_focus=true#step:4:17).

File: src/utils.ts [hunk 4]
```diff
@@ -5,6 +5,22 @@ export interface BoundingBox {
...
+  boxSizing?: string;
+  cursor?: string;
+  position?: string;
+  zIndex?: string;
+};
+
+export type ElementOverlayOptions = {
+  className?: string;
+  style?: ElementOverlayStyleOptions;
+};
```
Comment (hmarr) [inline src/utils.ts:23]: Style nit: would you mind adding a blank line between the type declaration and the function below?

File: src/utils.ts (hunk 1 above)
Comment (hmarr) [reply src/utils.ts:9]: Thanks, this looks better. There are a few other places that need updating too, though.

**Comment by hmarr** [comment]
Thanks for the contribution @ahmetsaridogan!

**Pull request merged**